### ISSUE TITLE
nan loss in MuLaN training

### ISSUE THREAD

**Issue by ukemamaster** [lucidrains/musiclm-pytorch#20]
@lucidrains 
While training `MuLaN` on a dataset of around 5.2k samples, the loss goes to `nan` after some 15-16k steps. 
My batch size is 4, and the `text` part of the data samples are tokenized using: 

```python
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
text_in_numbers = tokenizer.encode(text)
```
Does it has something to do with the zero division? or square-root of 0 in the loss function?

**Comment by lucidrains** [comment]
it could be a number things, but definitely overtrained

also, for contrastive learning, a batch size of 4 is too small. even 64 is too small

yeah, this can only realistically be done over at open clip

**Comment by lucidrains** [comment]
how do your loss curves look before it diverged?

**Comment by ukemamaster** [comment]
@lucidrains Thisi is how the loss looks like: 

![mulan_loss](https://user-images.githubusercontent.com/59258087/219595358-52a90443-17e7-4a44-9e67-8695bcd72265.png)

After some 20k steps, it becomes `inf`, and after a few more steps it becomes `nan`. The plot looks very strange to me. 

Regarding the batch size, i tried to increase but i get memory error like [this](https://github.com/lucidrains/musiclm-pytorch/issues/8#issuecomment-1434160845). 
Also, i am only feeding 5.5 seconds of audio (at 16kHz) i.e., 88k samples to the model. If i increase this, i get the same memory error.

**Comment by lucidrains** [comment]
@ukemamaster (1) that's very abnormal haha (2) yeah, you need a lot of memory (3) i have an issue at open clip and an active PR that i'm working on, feel free to subscribe!

**Comment by ukemamaster** [comment]
@lucidrains [This](https://github.com/mlfoundations/open_clip/pull/406#issue-1570397463) is the PR you ar atalking about, right? 
If it gets aaccepted, then what? Will we still need to train mulan? or we will get a pretrained one, ready to use?

**Comment by lucidrains** [comment]
@ukemamaster yes, that's the one. it will have to be trained and then served as a pretrained model. i'll write a wrapper to use it within this repository, similar to [here](https://github.com/lucidrains/DALLE2-pytorch/blob/main/dalle2_pytorch/dalle2_pytorch.py#L380). you should get involved; there's many researchers and startups interested in this

**Comment by deepak-newzera** [comment]
@lucidrains I am also waiting for a pre-trained model, for immediate use. Please make it available asap. Thank you.

**Comment by Mingxiangyu** [comment]
> @ukemamaster yes, that's the one. it will have to be trained and then served as a pretrained model. i'll write a wrapper for use within this repository, similar to [here](https://github.com/lucidrains/DALLE2-pytorch/blob/main/dalle2_pytorch/dalle2_pytorch.py#L380). you should get involved; there's many researchers and startups interested in this @ukemamaster 是的，就是这个。它将必须被训练，然后被用作预训练的模型。我将编写一个在这个存储库中使用的包装器，类似于这里。你应该参与进来有很多研究人员和创业公司对此感兴趣

@lucidrains Thank you for sharing and looking forward to your progress

**Comment by skychwang** [comment]
to add on to this, using a batch size of 512, similar loss curve -> nan loss behavior,
